pkg/svc/extension.go

```diff
@@ -288,28 +291,33 @@ func (s *extensionRegistrySvc) ListExtensions(ctx context.Context, query *extens
 func (s *extensionRegistrySvc) UpdateExtension(ctx context.Context, req *extension.Extension) (res *extension.Extension, err error) {
 	s.logger.Print("extension.updateExtension")
 	ext := extensionToDomain(req)
-	extRecord, err := s.registry.GetExtension(ctx, ext.ID, false)
+	extension, err := s.registry.GetExtension(ctx, ext.ID)
 	if err != nil {
 		return nil, errToRest(err)
 	}
 	// update only attributes present in the update request
 	extUpdate := domain.Extension{
 		ID:            ext.ID,
-		Product:       util.DerefString(req.Product, extRecord.Product),
-		Version:       util.DerefString(req.Version, extRecord.Version),
-		Description:   util.DerefString(req.Description, extRecord.Description),
-		Zone:          util.DerefString(req.Zone, extRecord.Zone),
-		Configuration: extRecord.Configuration,
+		Product:       util.DerefString(req.Product, extension.Product),
+		Version:       util.DerefString(req.Version, extension.Version),
+		Description:   util.DerefString(req.Description, extension.Description),
+		Zone:          util.DerefString(req.Zone, extension.Zone),
+		Configuration: extension.Configuration,
+		Services:      extension.Services,
 	}
-	if req.Configuration != nil {
+	if req.Configuration != nil && len(req.Configuration) > 0 {
 		extUpdate.Configuration = req.Configuration
 	}
 
+	if req.Services != nil && len(req.Services) > 0 {
```

Comment:
just to confirm that this, unlike the change above concerning the `Configuration` map, is acceptable semantically, because it doesn't make sense for a user to set the service list to an empty list and even if it would, there is still the alternative of deleting services one by one.